home/app_manager.py

```diff
@@ -234,17 +253,64 @@ def _refresh_prereleases(self, change):
                 self.include_prereleases.value or prerelease_installed
             )
 
+    def find_unmatched_requirements(self, app_version, python_bin):
+        """Calling find_incompatibilities and get unmatched requirements"""
+        return [
+            r[1] for r in self.app._app.find_incompatibilities(app_version, python_bin)
+        ]
+
+    def find_to_be_installed_dependencies(self, app_version, python_bin=None):
+        """return a list of Dependency"""
+        if python_bin is None:
+            python_bin = sys.executable
+
+        if app_version is None:
+            return []
+
+        dependencies = []
+        unmatched_requirements = self.find_unmatched_requirements(
+            app_version, python_bin
+        )
+        packages = find_installed_packages(python_bin)
+        for requriment in unmatched_requirements:
+            installed = "n/a"
+
+            for p in packages:
+                if requriment.name == p.name:
+                    installed = f"{p.name}=={p.version}"
+
+            dependencies.append(Dependency(installed, str(requriment)))
+
+        return dependencies
+
+    def _check_strict_dependencies(self, dependencies):
+        """Check dependencies that are not allowed to be override
+        return True if the strict dependencies satisfied
+
+        Note: Only check aiida-core compatibility at the moment
+        """
+        for dep in dependencies:
+            if "aiida-core" in dep.installed:
```

Comment:
Please generalize this code for the future, define a list of strict dependencies as a global constant (probably at the top of the file or somwhere).

I also wonder, should this function be here or rather in `aiidalab` repo? I don't have an answer now, but something to think about.